lib/utils.js

```diff
@@ -96,6 +96,9 @@ function patchEventTargetMethods(obj) {
   // This is required for the addEventListener hook on the root zone.
   obj[keys.common.addEventListener] = obj.addEventListener;
   obj.addEventListener = function (eventName, handler, useCapturing) {
+    if (!handler || (typeof(handler) !== 'function' && typeof(handler.handleEvent) !== 'function')) {
+      return;
```

Comment:
I'm not sure what do you mean...If i kept the first condition only then my test would fail...And it's not in line with most people's expectations. See https://github.com/angular/zone.js/issues/167  

And as you said
> From MDN: "Calling removeEventListener() with arguments that do not identify any currently registered EventListener on the EventTarget has no effect."

I think it's the same as addEventListener.